src/Gui/View3DSettings.cpp

```diff
@@ -337,6 +340,15 @@ void View3DSettings::OnChange(ParameterGrp::SubjectType &rCaller,ParameterGrp::M
             _viewer->setEnabledNaviCube(rGrp.GetBool("ShowNaviCube", true));
         }
     }
+    else if (strcmp(Reason,"AxisXColor") == 0 || strcmp(Reason,"AxisYColor") == 0 || strcmp(Reason,"AxisZColor") == 0) {
+        for (auto _viewer : _viewers) {
+            if(_viewer->hasAxisCross()) {
+                _viewer->setAxisCross(false);  // Force redraw
+                _viewer->setAxisCross(true);
+            }
```

Comment:
out of curiosity, why is this needed _before_ updating the colors?

Reply:
`View3DInventorViewer::updateColors` updates color for locally implemented cross and navicube. These are drawn on top of the 3D view:
![image](https://github.com/user-attachments/assets/0aa27f64-ea5a-4440-b024-613f17a265b5)
![image](https://github.com/user-attachments/assets/1e91f63a-95bb-4c77-97a2-583f748c0c69)

`View3DInventorViewer::setAxisCross` forces re-render of globally implemented `SoAxisCrossKit`:
![image](https://github.com/user-attachments/assets/4390e9e7-9639-410b-b020-646f0fb64057)

In other words; `View3DInventorViewer::updateColors` and `View3DInventorViewer::setAxisCross` affects different objects, and therefore the order is not of importance. Though I should probably give `SoAxisCrossKit` it's own `updateColors` method for consistent implementation

Reply:
that would be ideal for code clarity